charmhelpers/contrib/network/ovs/__init__.py

```diff
@@ -455,10 +483,14 @@ def get_certificate():
         return None
 
 
-def check_for_eni_source():
-    ''' Juju removes the source line when setting up interfaces,
-    replace if missing '''
+def setup_eni():
+    """Makes sure /etc/network/interfaces.d/ exists and will be parsed.
 
+    Indeed when setting up interfaces, Juju removes from
+    /etc/network/interfaces the line sourcing interfaces.d/
+    """
+    if not os.path.exists('/etc/network/interfaces.d'):
+        os.makedirs('/etc/network/interfaces.d', mode=0o755)
     with open('/etc/network/interfaces', 'r') as eni:
         for line in eni:
             if line == 'source /etc/network/interfaces.d/*':
```

Comment:
I know it's not part of this review, but this search feels really brittle!  Why not something like:

     re.search(r"^source\s+/etc/network/interfaces.d/\*\s*$")

Which would ignore space issues? (obviously, I've not actually tested that regex, but it would be something like that).